playground/publishers/Publishers.AppHost/Program.cs

```diff
@@ -1,17 +1,30 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#pragma warning disable ASPIREAZURE001
 #pragma warning disable ASPIREPUBLISHERS001
 
 var builder = DistributedApplication.CreateBuilder(args);
 
-builder.AddAzureContainerAppEnvironment("env");
-
-builder.AddDockerComposeEnvironment("docker-env");
-
-builder.AddKubernetesPublisher();
-
-builder.AddAzurePublisher("azure");
+if (builder.ExecutionContext.PublisherName == "azure" ||
+    builder.ExecutionContext.IsInspectMode)
+{
+    builder.AddAzureContainerAppEnvironment("env");
+}
+
+if (builder.ExecutionContext.PublisherName == "docker-compose" ||
+    builder.ExecutionContext.IsInspectMode)
+{
+    builder.AddDockerComposeEnvironment("docker-env");
+}
+
+if (builder.ExecutionContext.PublisherName == "kubernetes" ||
+    builder.ExecutionContext.IsInspectMode)
+{
+    builder.AddKubernetesEnvironment("k8s-env");
+}
+
+builder.AddAzurePublisher();
```

Comment:
Why's this necessary? Doesn't `AddAzureContainerAppEnvironment` bring it in?

Reply:
Not yet, let me change it.